Apps/WebClient/src/Server/Services/WalletService.cs

```diff
@@ -281,9 +281,101 @@ public async Task<RequestResult<IEnumerable<WalletCredentialModel>>> CreateCrede
         }
 
         /// <inheritdoc/>
-        public Task<RequestResult<WalletConnectionModel>> DisconnectConnection(Guid connectionId, string hdId)
+        public async Task<RequestResult<WalletConnectionModel>> DisconnectConnectionAsync(Guid connectionId, string hdId)
         {
-            throw new NotImplementedException();
+            this.logger.LogDebug($"Disconnecting wallet connection. getting connection from database. connectionId: {connectionId} user: {hdId}");
+            DBResult<WalletConnection> dbResult = this.walletDelegate.GetConnection(connectionId, hdId);
+            if (dbResult.Status != DBStatusCode.Read)
+            {
+                this.logger.LogDebug($"Error disconnecting wallet connection. connection not found. connectionId: {connectionId} user: {hdId}. {JsonSerializer.Serialize(dbResult)}");
+                return new RequestResult<WalletConnectionModel>()
+                {
+                    ResultStatus = ResultType.Error,
+                    ResultError = new RequestResultError()
+                    {
+                        ResultMessage = "Error disconnecting wallet connection. connection not found",
+                    },
+                };
+            }
+
+            if (dbResult.Payload.Status != WalletConnectionStatus.Connected)
```

Comment:
I think that Disconnect is logically deleting the connection and if the client has lost the request they should still be able to remove it in the Pending state which is similar o how Revoke on Credential works.

Reply:
Agreed.